java/download-preplanned-map-area/src/main/java/com/esri/arcgisruntime/sample/downloadpreplannedmaparea/MainActivity.java

```diff
@@ -0,0 +1,392 @@
+                // add it to the map view
+                mMapView.setMap(offlineMap);
+                // add the map name to the list view of downloaded map areas
+                mDownloadedMapAreaNames.add(offlineMap.getItem().getTitle());
+                // select the downloaded map area
+                mDownloadedMapAreasListView.setItemChecked(mDownloadedMapAreaNames.size() - 1, true);
+                mDownloadedMapAreasAdapter.notifyDataSetChanged();
+                // de-select the area in the preplanned areas list view
+                mPreplannedAreasListView.clearChoices();
+                mPreplannedMapAreasAdapter.notifyDataSetChanged();
+                // add the offline map to a list of downloaded map areas
+                mDownloadedMapAreas.add(offlineMap);
+                // hide the area of interest graphics
+                mAreasOfInterestGraphicsOverlay.setVisible(false);
+                // disable the download button
+                mDownloadButton.setEnabled(false);
+              } else {
+                // collect the layer and table errors into a single alert message
+                StringBuilder stringBuilder = new StringBuilder("Errors: ");
+                Map<Layer, ArcGISRuntimeException> layerErrors = downloadPreplannedOfflineMapResult.getLayerErrors();
+                for (Map.Entry<Layer, ArcGISRuntimeException> layer : layerErrors.entrySet()) {
+                  stringBuilder.append("Layer: ").append(layer.getKey().getName()).append(". Exception: ")
+                      .append(layer.getValue().getMessage()).append(". ");
+                }
+                Map<FeatureTable, ArcGISRuntimeException> tableErrors = downloadPreplannedOfflineMapResult
+                    .getTableErrors();
+                for (Map.Entry<FeatureTable, ArcGISRuntimeException> table : tableErrors.entrySet()) {
+                  stringBuilder.append("Table: ").append(table.getKey().getTableName()).append(". Exception: ")
+                      .append(table.getValue().getMessage()).append(". ");
+                }
+                String error = "One or more errors occurred with the Offline Map Result: " + stringBuilder;
+                Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+                Log.e(TAG, error);
+              }
+            } else {
+              String error = "Job finished with an error: " + mDownloadPreplannedOfflineMapJob.getError();
+              Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+              Log.e(TAG, error);
+            }
+          });
+        } catch (InterruptedException | ExecutionException e) {
+          String error = "Failed to generate default parameters for the download job: " + e.getCause().getMessage();
+          Toast.makeText(this, error, Toast.LENGTH_LONG).show();
+          Log.e(TAG, error);
+        }
+      });
+    }
+  }
+
+  private void createPreplannedAreasListView(ArcGISMap onlineMap, OfflineMapTask offlineMapTask) {
```

Comment:
I think there's a lot here that could be wrapped into a custom adapter. Unless you're wanting to not go down this route.

Reply:
Yeah trying to avoid custom adapter if I can

Reply:
Fair enough. Will approve.